packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -580,17 +581,28 @@ import io.flutter.embedding.engine.plugins.shim.ShimPluginRegistry;
  */
 @Keep
 public final class GeneratedPluginRegistrant {
+  private static final String TAG = "GeneratedPluginRegistrant";
   public static void registerWith(@NonNull FlutterEngine flutterEngine) {
 {{#needsShim}}
     ShimPluginRegistry shimPluginRegistry = new ShimPluginRegistry(flutterEngine);
 {{/needsShim}}
 {{#plugins}}
   {{#supportsEmbeddingV2}}
-    flutterEngine.getPlugins().add(new {{package}}.{{class}}());
+    try {
+      flutterEngine.getPlugins().add(new {{package}}.{{class}}());
+    } catch(Exception e) {
+      Log.e(TAG, "Error registering plugin {{name}}, {{package}}.{{class}} " +
+      "The app might not function as expected unless you remove it from pubspec.yaml.", e);
```

Comment:
I'd avoid giving an official recommendation here since it may be "fixable" depending on the usage of the plugin itself. Just keeping the first sentence is fine.